examples/03_better_tool/ex_better_tool/main.rb

```diff
@@ -0,0 +1,183 @@
+# Copyright 2021 Trimble Inc
+# Licensed under the MIT license
+
+require 'sketchup.rb'
+
+module Examples
+  module BetterTool
+
+    class LineTool
+
+      # Threshold in logical screen pixels for when the mouse is considered to
+      # be dragged.
+      APERTURE = 10
+
+      def activate
+        @mouse_ip = Sketchup::InputPoint.new
+        @picked_first_ip = Sketchup::InputPoint.new
+        @x_down = @y_down = nil
+        update_ui
+      end
+
+      def deactivate(view)
+        view.invalidate
+      end
+
+      def enableVCB?
+        picked_first_point?
+      end
+
+      def resume(view)
+        update_ui
+        view.invalidate
+      end
+
+      def onCancel(reason, view)
+        reset_tool
+        view.invalidate
+      end
+
+      def onMouseMove(flags, x, y, view)
+        if picked_first_point?
+          @mouse_ip.pick(view, x, y, @picked_first_ip)
+          Sketchup.vcb_value = @mouse_ip.position.distance(@picked_first_ip.position)
```

Comment:
VCB could be reset during orbit. I'd think it's better to cache the distance to an instance variable and set it as part of `update_ui`.

Reply:
Done